io/plugins/eu.esdihumboldt.hale.io.gml.geometry/src/eu/esdihumboldt/hale/io/gml/geometry/handler/GenericGeometryHandler.java

```diff
@@ -236,23 +240,23 @@ else if (!commonCrs.equals(crs)) {
 				// create a MultiPolygon
 				Polygon[] polygons = new Polygon[geomList.size()];
 				for (int i = 0; i < geomList.size(); i++) {
-					polygons[i] = (Polygon) geomList.get(i);
+					polygons[i] = movePolygonToUniversalGrid((Polygon) geomList.get(i), reader);
```

Comment:
The geometries that are combined here (Polygons, Points, etc.) have been created somewhere else (other handlers) where they should have been already moved. We're not actually creating geometries from coordinates that we are reading here. So I think any adaptions to the geometries themselves can be omitted.